src/hooks/staticContent.ts

```diff
@@ -82,7 +82,8 @@ export const useStaticContent = <T>({
         const json = data?.publicJsonFile?.content;
 
         return parseFromJson ? parseFromJson(json) : json;
-      } else {
+      } else if (data || queryError) {
+        // set error true if there is bad data without `publicJsonFile.content` or some `queryError`
```

Comment:
Good point. It really makes sense to have some check here. Didn't see that in the review of the first PR.

However as the original issue was to check if there is data, I'm a little unsure if it is smart to check for data.
Wouldn't it make more sense to check for `loading` instead? Like `!loading || queryError`?

Reply:
at this particular point, we know that `data` is just correct, if it has `publicJsonFile.content`, otherwise it wouldn't be a correct static content response from the server. this means, if there is other `data` with some random information, we can handle it as error.
but i guess considering `loading` here in addition makes sense, so that we would not have an truthy error state if loading - i will change to: `!loading && data` to not have `error: true, loading: true` possible at the same time in the hook return. in addition i will remove `|| queryError` as this is already considered in the hook return for `error`.